JavaScript-Runtime-(JSRT)-Overview.md

```diff
@@ -85,9 +85,117 @@ All other supported ES2015 features do not require any setup to run in JSRT.
+JsCreateFunction(LogCB, nullptr, &logFunc);
+JsGetPropertyIdFromName(L"log", &logPropId);
+JsSetProperty(console, logPropId, logFunc, true);
+// set console as property of global object
+JsGetGlobalObject(&global);
+JsGetPropertyIdFromName(L"console", &consolePropId);
+JsSetProperty(global, consolePropId, console, true);
+```
+
+Now in JavaScript, you can do,
+```js
+console.log('Hello world');
+```
+
+If you need to expose an object that hosts data not directly exposed to JS, you can create an external object via `JsCreateExternalObject`.
+
+### Call JavaScript functions from native
+
+You can call JavaScript functions by calling `JsRun/JsRunScript`, or alternatively you can get the native reference of your JS function and then use `JsCallFunction` to call it. For example, to call `func` defined in global scope in JS,
+
+```cpp
+JsValueRef func, funcPropId, global, undefined, result;
+JsGetGlobalObject(&global);
+JsGetUndefinedValue(&undefined);
+JsValueRef args[] = { undefined };
+JsGetPropertyIdFromName(L"func", &funcPropId);
+JsGetProperty(global, funcPropId, &func);
+// note that args[0] is thisArg of the call; actual args start at index 1
+JsCallFunction(func, args, 1, &result);
+``` 
+
+### Script serialization with lazy source loading  
+
+To support startup and execution efficiency, JSRT APIs provide the capability of pre-parsing, generating syntax trees and caching the bytecode for scripts. It is especially useful for server scenarios where the same script could get executed for thousands of times.
+
+The basic idea is to use variants of `JsSerializeScript` to parse and store the bytecode and then use `JsParse/RunSerializedScript` to parse or run the bytecode. Source is typically not needed for running serialized code, therefore you can also lazy-load the source script using `JsRunSerialized/JsRunSerializedScriptWithCallback`.
+
+
+```cpp
+// serialize script and cache the bytecode on disk
+void serializeScript(const wchar_t *script, const wchar_t *bytecodeBufferPath) {
+    JsValueRef result;
+    unsigned long bufferSize = 0;
+    BYTE *buffer = nullptr;
+    JsSerializeScript(script, buffer, &bufferSize); // get correct buffer size
+    buffer = new BYTE[bufferSize];
+    JsSerializeScript(script, buffer, &bufferSize); // actual serialization
+    bytecodeCacheMap[scriptPath] = 
+    SaveByteCode(buffer, bytecodecachePath);
+}
+
+// load and run bytecode 
+void runSerializeScript(const wchar_t *bytecodeBufferPath, const wchar_t *scriptBufferPath) {
+    JsValueRef result;
+    BYTE *buffer = LoadByteCode(bytecodeBufferPath);
+    SerializedSourceContext* context = new SerializedSourceContext();
+    context->scriptPath = scriptBufferPath;
+    JsRunSerializedScriptWithCallback(
+        [](JsSourceContext sourceContext, const WCHAR** scriptBuffer)
+    {
+        // load script when necessary
+        *scriptBuffer = LoadScript(((SerializedSourceContext*)sourceContext)->scriptPath);
+        return true;
+    },
```

Comment:
Should this whole block (starting from open brace line 186) be indented one level deeper (to line up with `[]` starting the lambda)?